### ISSUE TITLE
Fails to locate game on secondary drive

### ISSUE THREAD

**Issue by GeckoEidechse** [WilliamVenner/steamlocate-rs#37]
[My application](https://github.com/R2NorthstarTools/FlightCore/) uses `steamlocate-rs` to find the install directory of [Titanfall2](https://store.steampowered.com/app/1237970/Titanfall_2/) (`1237970`).

A user of my application on Fedora 38 noticed however that it was unable to locate their install of Titanfall2.
On their setup they installed Steam via the native Fedora package.

As such their library folders are located in

```
~/.local/share/Steam/config/libraryfolders.vdf
~/.local/share/Steam/steamapps/libraryfolders.vdf
```

which (partially anonymised) looks like:

```
"libraryfolders"
{
    "0"
    {
        "path"        "/home/XXXXXXXXX/.local/share/Steam"
        "label"        ""
        "contentid"        "XXXXXXXXXXXXXXXXXXXX"
        "totalsize"        "0"
        "update_clean_bytes_tally"        "6504277467"
        "time_last_update_corruption"        "0"
        "apps"
        {
            "XXXXXXX"        "XXXXXXXXXX"
        }
    }
    "1"
    {
        "path"        "/mnt/0678788f-1ffe-4a12-8608-8ef3b8467106/SteamLibrary"
        "label"        ""
        "contentid"        "4113662200414089328"
        "totalsize"        "737229742080"
        "update_clean_bytes_tally"        "10119163182"
        "time_last_update_corruption"        "0"
        "apps"
        {
            "XXXXXXXXX"        "XXXXXXXXX"
            "1237970"        "68365272890"
            "XXXXXXXXX"        "XXXXXXXXX"
            "XXXXXXXXX"        "XXXXXXXXX"
        }
    }
}
```

Unfortunately I have not yet been able to figure out where the game detection fails (reading `libraryfolders.vdf`, accessing secondary drive, ...)

For detection I'm using [the newest release of steamlocate, i.e. `1.2.1`](https://github.com/R2NorthstarTools/FlightCore/blob/e181f6b7121738d589bdd47e38736ed5978939f5/src-tauri/Cargo.lock#L3787-L3799)

**Comment by CosmicHorrorDev** [comment]
Can you reproduce on the `2.0.0-alpha.0` release? It's possible there is a value in their app manifest that fails to parse on the `1.2.1` release

**Comment by GeckoEidechse** [comment]
Aight, I'll try to push out a build with that version of the library over the next day or so and get them to test it to see if it solves it \^\^

**Comment by GeckoEidechse** [comment]
Ok, didn't get to push out a build yet but talking to the affect person again, it turns out they still had leftover files from a Flatpak install of Steam and `steamlocate` apparently was prioritising those and ignoring the other files?

Still trying to find out more about this...

**Comment by GeckoEidechse** [comment]
Yeah, so to confirm, removing the leftover Flatpak Steam files, fixed the issue for the affected person.

**Comment by mainrs** [comment]
Maybe this is useful in some way. These seem to be the installation directories of Steam on various Linux distros:

https://github.com/lutris/lutris/blob/73ea627dca06e049dcbb57222932cb82227ea576/lutris/util/steam/config.py#L13C5-L25
